week2/project/ecommerce/src/ProductDetail.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+const ProductDetail = () => {
+  const { id } = useParams();
+  const [product, setProduct] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const fetchProduct = async () => {
+      try {
+        const response = await fetch(`https://fakestoreapi.com/products/${id}`);
+        if (!response.ok) throw new Error("Failed to fetch product details");
+        const data = await response.json();
+        setProduct(data);
+      } catch (err) {
+        setError(err.message);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchProduct();
+  }, [id]);
+
+  if (loading) return <p>Loading product...</p>;
+  if (error) return <p>{error}</p>;
+
+  return (
+    <div className="product-detail">
+      <img src={product.image} alt={product.title} className="product-image" />
+      <div className="product-info">
+        <h2 className="product-title">{product.title}</h2>

```

Comment:
```suggestion
        <h1 className="product-title">{product.title}</h1>

```

Since ProductDetail is a new page, it is better to use h1